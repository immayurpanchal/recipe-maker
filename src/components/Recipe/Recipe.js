import React from 'react';

const Recipe = (props) => {
	const SIZE = '556x370';
	const TYPE = 'jpg';
	return (
		<div className='recipe'>
			<figure className='recipe__fig'>
				<img
					src={props && props.recipe && `https://spoonacular.com/recipeImages/${props.recipe.id}-${SIZE}.${TYPE}`}
					alt={props && props.recipe && props.recipe.title}
					className='recipe__img'
				/>
				<h1 className='recipe__title'>
					<span>{props && props.recipe && props.recipe.title}</span>
				</h1>
			</figure>
			<div className='recipe__details'>
				<div className='recipe__info'>
					<svg className='recipe__info-icon'>
						<use href='img/icons.svg#icon-stopwatch' />
					</svg>
					<span className='recipe__info-data recipe__info-data--minutes'>
						{props && props.recipe && props.recipe.readyInMinutes}
					</span>
					<span className='recipe__info-text'> minutes</span>
				</div>
				<div className='recipe__info'>
					<svg className='recipe__info-icon'>
						<use href='img/icons.svg#icon-man' />
					</svg>
					<span className='recipe__info-data recipe__info-data--people'>
						{props && props.recipe && props.recipe.servings}
					</span>
					<span className='recipe__info-text'> servings</span>

					<div className='recipe__info-buttons'>
						<button className='btn-tiny'>
							<svg>
								<use href='img/icons.svg#icon-circle-with-minus' />
							</svg>
						</button>
						<button className='btn-tiny'>
							<svg>
								<use href='img/icons.svg#icon-circle-with-plus' />
							</svg>
						</button>
					</div>
				</div>
				<button className='recipe__love'>
					<svg className='header__likes'>
						<use href='img/icons.svg#icon-heart-outlined' />
					</svg>
				</button>
			</div>

			<div className='recipe__ingredients'>
				<ul className='recipe__ingredient-list'>
					{props &&
						props.recipe &&
						props.recipe.ingredients &&
						props.recipe.ingredients.map(({ name, amount: { us: { value, unit } } }) => {
							return (
								<li className='recipe__item'>
									<svg className='recipe__icon'>
										<use href='img/icons.svg#icon-check' />
									</svg>
									<div className='recipe__count'>{value}</div>
									<div className='recipe__ingredient'>
										<span className='recipe__unit'>{unit} </span>
										{name}
									</div>
								</li>
							);
						})}
				</ul>

				<button className='btn-small recipe__btn'>
					<svg className='search__icon'>
						<use href='img/icons.svg#icon-shopping-cart' />
					</svg>
					<span>Add to shopping list</span>
				</button>
			</div>

			<div className='recipe__directions'>
				<h2 className='heading-2'>How to cook it</h2>
				<p className='recipe__directions-text'>
					This recipe was carefully designed and tested by
					<span className='recipe__by'>The Pioneer Woman</span>. Please check out directions at their website.
				</p>
				<a className='btn-small recipe__btn' href='http://thepioneerwoman.com/cooking/pasta-with-tomato-cream-sauce/'>
					<span>Directions</span>
					<svg className='search__icon'>
						<use href='img/icons.svg#icon-triangle-right' />
					</svg>
				</a>
			</div>
		</div>
	);
};

export default Recipe;
