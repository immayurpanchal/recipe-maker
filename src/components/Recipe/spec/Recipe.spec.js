import React from 'react';
import { render } from '@testing-library/react';
import Recipe from '../Recipe';

describe('Header component', () => {
	it('Render the component', () => {
		const { container } = render(<Recipe />);
		expect(container).toMatchSnapshot();
	});

	it('Pass the props', () => {
		const recipe = {
			title: '',
			readyInMinutes: '',
			servings: '',
			ingredients: [
				{
					name: 'test',
					amount: {
						us: { value: 20, unit: 41 },
					},
					id: 2,
				},
			],
		};

		const { container } = render(
			<Recipe recipe={recipe} onAddToShoppingList={() => jest.fn()} onLiked={() => jest.fn()} />
		);

		expect(container).toMatchSnapshot();
	});
});
