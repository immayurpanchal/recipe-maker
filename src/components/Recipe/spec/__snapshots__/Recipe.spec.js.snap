// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Pass the props 1`] = `
<div>
  <div
    class="recipe"
  >
    <figure
      class="recipe__fig"
    >
      <img
        alt=""
        class="recipe__img"
        src="https://spoonacular.com/recipeImages/undefined-556x370.jpg"
      />
      <h1
        class="recipe__title"
      >
        <span />
      </h1>
    </figure>
    <div
      class="recipe__details"
    >
      <div
        class="recipe__info"
      >
        <svg
          class="recipe__info-icon"
        >
          <use
            href="img/icons.svg#icon-stopwatch"
          />
        </svg>
        <span
          class="recipe__info-data recipe__info-data--minutes"
        />
        <span
          class="recipe__info-text"
        >
           minutes
        </span>
      </div>
      <div
        class="recipe__info"
      >
        <svg
          class="recipe__info-icon"
        >
          <use
            href="img/icons.svg#icon-man"
          />
        </svg>
        <span
          class="recipe__info-data recipe__info-data--people"
        />
        <span
          class="recipe__info-text"
        >
           servings
        </span>
      </div>
      <button
        class="recipe__love"
      >
        <svg
          class="header__likes"
        >
          <use
            href="img/icons.svg#icon-heart-outlined"
          />
        </svg>
      </button>
    </div>
    <div
      class="recipe__ingredients"
    >
      <ul
        class="recipe__ingredient-list"
      >
        <li
          class="recipe__item"
        >
          <svg
            class="recipe__icon"
          >
            <use
              href="img/icons.svg#icon-check"
            />
          </svg>
          <div
            class="recipe__count"
          >
            20
          </div>
          <div
            class="recipe__ingredient"
          >
            <span
              class="recipe__unit"
            >
              41
               
            </span>
            test
          </div>
        </li>
      </ul>
      <button
        class="btn-small recipe__btn"
      >
        <svg
          class="search__icon"
        >
          <use
            href="img/icons.svg#icon-shopping-cart"
          />
        </svg>
        <span>
          Add to shopping list
        </span>
      </button>
    </div>
    <div
      class="recipe__directions"
    >
      <h2
        class="heading-2"
      >
        How to cook it
      </h2>
      <p
        class="recipe__directions-text"
      >
        This recipe was carefully designed and tested by
        <span
          class="recipe__by"
        >
          The Pioneer Woman
        </span>
        . Please check out directions at their website.
      </p>
      <a
        class="btn-small recipe__btn"
        href="http://thepioneerwoman.com/cooking/pasta-with-tomato-cream-sauce/"
      >
        <span>
          Directions
        </span>
        <svg
          class="search__icon"
        >
          <use
            href="img/icons.svg#icon-triangle-right"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`Header component Render the component 1`] = `
<div>
  <div
    class="recipe"
  />
</div>
`;
